src/Axe/Managers/ICompilerManager.cs

```diff
@@ -16,21 +16,14 @@ public interface ICompilerManager
         /// Returns a list of code block available for current user
         /// </summary>
         /// <returns>List of CodeBlockVm</returns>
-        Task<List<CodeBlockVm>> GetCodeBlocks(int technologyId);
+        Task<List<T>> GetCodeBlocks<T>(int technologyId) where T : class;
```

Comment:
GetCodeBlocks cannot possibly return any class, here should a be constraint on `T`, related to CodeBlockVm